client/extensions/woocommerce/state/ui/shipping/zones/README.md

```diff
@@ -0,0 +1,77 @@
+Shipping Zones
+===============
+
+This module is used to manage Shipping Zones settings. This includes adding, removing and editing shipping zones. Since
+shipping zones have a lot of information, parts of them will be handled by other modules inside this one, such as
+the shipping methods a Zone has configured, or the locations corresponding to a Zone.
+
+## Reducer
+
+In this part of the state only will be stored the "differences" between what's on the server and what the user has configured.
+So, for example, the initial state would look like this:
+```js
+{
+	zones: {
+		creates: [], // No new zones have been created
+		updates: [], // No settings for any given zone have been changed
+		deletes: [], // No zones have been deleted
+		currentlyEditingId: null, // The "Edit Zone" modal isn't open
+	}
+}
+```
+
+After the user has made some changes, the state could look like this:
+```js
+{
+	zones: {
+		creates: [
+			{ // 1 new zone has been created:
+				id: { index: 0 }, // Temporary ID, since this zone still doesn't have a real ID provided by the API
```

Comment:
Basically to differentiate it from a "regular" ID. I guess we could have chosen something different like negative numbers or something like that. When calling `state/ui/helpers/nextBucketIndex()` with the `creates` collection, it will return a new unique ID object.

It took a while to sink in for me, I suggest looking at a simpler UI reducer like the `ui/products/edits-reducer.js`. Also, we should probably document better this whole deal.